--- runConfigureICU	2017-06-23	03:38:28
+++ runConfigureICU	2017-06-24	18:04:13
@@ -28,6 +28,7 @@
 Options: -h, --help         Print this message and exit
          --enable-debug     Enable support for debugging
          --disable-release  Disable presetting optimization flags
+         --static-runtime   Use a statically linked C/C++ runtime (Cygwin/MSVC and MinGW only)
 
 If you want to add custom CFLAGS or CXXFLAGS or similar, provide them _before_
 the runConfigureICU command:
@@ -72,6 +73,7 @@
 platform=
 debug=0
 release=1
+static_runtime=0
 
 while test $# -ne 0
 do
@@ -87,6 +89,9 @@
         release=0
         OPTS="$OPTS --disable-release"
         ;;
+    --static-runtime)
+        static_runtime=1
+        ;;
     *)
         platform="$1"
         shift
@@ -263,11 +268,16 @@
         THE_COMP="Microsoft Visual C++"
         CC=cl; export CC
         CXX=cl; export CXX
-        RELEASE_CFLAGS='-Gy -MD'
-        RELEASE_CXXFLAGS='-Gy -MD'
-        DEBUG_CFLAGS='-Zi -MDd'
-        DEBUG_CXXFLAGS='-Zi -MDd'
-        DEBUG_LDFLAGS='-DEBUG'
+        if [ "${static_runtime}" = "0" ]; then
+            RUNTIME_LINKER_OPTION='MD'
+        else
+            RUNTIME_LINKER_OPTION='MT'
+        fi
+        RUNTIME_LINKER_FLAG='/'${RUNTIME_LINKER_OPTION}
+        RELEASE_CFLAGS='/Gy '${RUNTIME_LINKER_FLAG}
+        RELEASE_CXXFLAGS='/Gy '${RUNTIME_LINKER_FLAG}
+        DEBUG_CFLAGS=${RUNTIME_LINKER_FLAG}d
+        DEBUG_CXXFLAGS=${RUNTIME_LINKER_FLAG}d
         ;;
     Cygwin/MSVC2005)
         THE_OS="Windows with Cygwin"
@@ -313,20 +323,32 @@
     MinGW)
         THE_OS="MinGW"
         THE_COMP="the GNU C++"
+        if [ "${static_runtime}" = "0" ]; then
+            RUNTIME_LINKER_FLAGS=''
+        else
+            RUNTIME_LINKER_FLAGS='-static-libgcc'
+        fi
         RELEASE_CFLAGS='-O3'
         RELEASE_CXXFLAGS='-O3'
+        LDFLAGS="${RUNTIME_LINKER_FLAGS}"
         export CXXFLAGS
+        export LDFLAGS
         ;;
     MSYS/MSVC)
         THE_OS="MSYS"
         THE_COMP="Microsoft Visual C++"
         CC=cl; export CC
         CXX=cl; export CXX
-        RELEASE_CFLAGS='-Gy -MD'
-        RELEASE_CXXFLAGS='-Gy -MD'
-        DEBUG_CFLAGS='-Zi -MDd'
-        DEBUG_CXXFLAGS='-Zi -MDd'
-        DEBUG_LDFLAGS='-DEBUG'
+        if [ "${static_runtime}" = "0" ]; then
+            RUNTIME_LINKER_OPTION='MD'
+        else
+            RUNTIME_LINKER_OPTION='MT'
+        fi
+        RUNTIME_LINKER_FLAG='/'${RUNTIME_LINKER_OPTION}
+        RELEASE_CFLAGS='/Gy '${RUNTIME_LINKER_FLAG}
+        RELEASE_CXXFLAGS='/Gy '${RUNTIME_LINKER_FLAG}
+        DEBUG_CFLAGS=${RUNTIME_LINKER_FLAG}d
+        DEBUG_CXXFLAGS=${RUNTIME_LINKER_FLAG}d
         ;;
     *BSD)
         THE_OS="BSD"
@@ -441,4 +463,4 @@
 else
 	echo $0: ./configure failed
 	exit 1
-fi
+fi
\ No newline at end of file
