--- ./qtbase/mkspecs/win32-g++/qmake.conf	2020-10-27 11:02:11 +0300
+++ ./qtbase/mkspecs/win32-g++/qmake.conf	2021-02-24 14:25:55 +0300
@@ -23,6 +23,8 @@
 QMAKE_LINK              = $${CROSS_COMPILE}g++
 QMAKE_LINK_C            = $${CROSS_COMPILE}gcc

+QMAKE_LFLAGS            = -static -static-libgcc -static-libstdc++
+
 QMAKE_CFLAGS_LTCG       = -flto
 QMAKE_CXXFLAGS_LTCG     = $$QMAKE_CFLAGS_LTCG
 QMAKE_LFLAGS_LTCG       = $$QMAKE_CFLAGS_LTCG
