--- ./qtbase/mkspecs/win32-g++/qmake.conf	2019-06-13 04:18:18 +0300
+++ ./qtbase/mkspecs/win32-g++/qmake.conf	2019-08-14 00:23:08 +0300
@@ -23,6 +23,8 @@
 QMAKE_LINK              = $${CROSS_COMPILE}g++
 QMAKE_LINK_C            = $${CROSS_COMPILE}gcc
 
+QMAKE_LFLAGS            = -static -static-libgcc -static-libstdc++
+
 QMAKE_CFLAGS_LTCG       = -flto
 QMAKE_CXXFLAGS_LTCG     = $$QMAKE_CFLAGS_LTCG
 QMAKE_LFLAGS_LTCG       = $$QMAKE_CFLAGS_LTCG
